opensearch_py_ml/ml_commons/model_access_control.py

```diff
@@ -0,0 +1,104 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+from typing import List, Optional
+
+from opensearchpy import OpenSearch
+
+from opensearch_py_ml.ml_commons.ml_common_utils import ML_BASE_URI
+from opensearch_py_ml.ml_commons.validators.model_access_control import (
+    validate_create_model_group_parameters,
+    validate_delete_model_group_parameters,
+    validate_search_model_group_parameters,
+    validate_update_model_group_parameters,
+)
+
+
+class ModelAccessControl:
+    API_ENDPOINT = "model_groups"
+
+    def __init__(self, os_client: OpenSearch):
+        self.client = os_client
+
+    def register_model_group(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        access_mode: Optional[str] = "private",
+        backend_roles: Optional[List[str]] = None,
+        add_all_backend_roles: Optional[bool] = False,
+    ):
+        validate_create_model_group_parameters(
+            name, description, access_mode, backend_roles, add_all_backend_roles
+        )
+
+        body = {"name": name, "add_all_backend_roles": add_all_backend_roles}
+        if description:
+            body["description"] = description
+        if access_mode:
+            body["access_mode"] = access_mode
+        if backend_roles:
+            body["backend_roles"] = backend_roles
+
+        return self.client.transport.perform_request(
+            method="POST", url=f"{ML_BASE_URI}/{self.API_ENDPOINT}/_register", body=body
+        )
+
+    def update_model_group(
+        self,
+        update_query: dict,
+        model_group_id: Optional[str] = None,
+        model_group_name: Optional[str] = None,
+    ):
+        validate_update_model_group_parameters(
+            update_query, model_group_id, model_group_name
+        )
+        if model_group_name:
```

Comment:
With client side validation, I don't think we need to do this check. ML-Commons will already check this. May be we can relay the message? Otherwise, we are doing a search index call, and ml-commons will also do another search index call. I don't think we need 2 same call.

Reply:
So, update model group on ML-Commons only support using a model group id. Here, we are giving a way to do this using the model group name

Reply:
For now, we made model group unique. But in the long run, we might not keep model group unique. That could break this functionality. So let's keep the experience same.

Reply:
So, I added a separate method which can act as utility to get the model id from the model name. These can be deprecated when model group isn't unique anymore.